src/js/components/stock-movement/inbound/StockMovementInboundTable.jsx

```diff
@@ -0,0 +1,399 @@
+import { confirmAlert } from 'react-confirm-alert';
+import {
+  RiDeleteBinLine,
+  RiDownload2Line,
+  RiInformationLine, RiPencilLine,
+} from 'react-icons/all';
+import { getTranslate } from 'react-localize-redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import Alert from 'react-s-alert';
+
+import { fetchShipmentStatusCodes, hideSpinner, showSpinner } from 'actions';
+import DataTable, { TableCell } from 'components/DataTable';
+import Button from 'components/form-elements/Button';
+import ActionDots from 'utils/ActionDots';
+import apiClient from 'utils/apiClient';
+import StatusIndicator from 'utils/StatusIndicator';
+import Translate, { translateWithDefaultMessage } from 'utils/Translate';
+
+import 'react-confirm-alert/src/react-confirm-alert.css';
+
+const StockMovementInboundTable = ({
+  filterParams,
+  translate,
+  fetchStatuses,
+  shipmentStatuses,
+  isShipmentStatusesFetched,
+  currentLocation,
+  history,
+  showTheSpinner,
+  hideTheSpinner,
+}) => {
+  const [tableData, setTableData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [pages, setPages] = useState(-1);
+  // Stored searching params for export case
+  const [currentParams, setCurrentParams] = useState({});
+  const [totalData, setTotalData] = useState(0);
+
+  const tableRef = useRef(null);
+
+  const fireFetchData = () => tableRef.current.fireFetchData();
+
+  // If filterParams change, refetch the data with applied filters
+  useEffect(() => {
+    fireFetchData();
+    if (!isShipmentStatusesFetched) fetchStatuses();
+  }, [filterParams]);
+
+  const exportStockMovements = () => {
+    apiClient.get('/openboxes/api/stockMovements', {
+      params: {
+        ...currentParams,
+        format: 'csv',
+      },
+      paramsSerializer: params => queryString.stringify(params),
+    })
+      .then((res) => {
+        const filename = res.headers['content-disposition'].split('filename="')[1].split('.')[0];
+        fileDownload(res.data, filename, 'text/csv');
+      });
+  };
+
+  const exportAllIncomingItems = () => {
+    apiClient.get('/openboxes/api/stockMovements/shippedItems', {
+      params: {
+        ...currentParams,
+        format: 'csv',
+      },
+      paramsSerializer: params => queryString.stringify(params),
+    })
+      .then((res) => {
+        const filename = res.headers['content-disposition'].split('filename="')[1].split('.')[0];
+        fileDownload(res.data, filename, 'text/csv');
+      });
+  };
+
+
+  const getStatusTooltip = status => translate(
+    `react.stockMovement.status.${status.toLowerCase()}.description.label`,
+    status.toLowerCase(),
+  );
+
+
+  const onFetchHandler = (state) => {
+    const offset = state.page > 0 ? (state.page) * state.pageSize : 0;
+    const sortingParams = state.sorted.length > 0 ?
+      {
+        sort: state.sorted[0].id,
+        order: state.sorted[0].desc ? 'desc' : 'asc',
+      } : undefined;
```

Comment:
I assume anyway backend sorts by something by default, but if not, add here sorting by date by default maybe?